### DISCUSSION TITLE
Please have standalone documentation of modes such as autoAction

### DISCUSSION THREAD

**Discussion by qwertie** [mobxjs/mobx#2920] [General]
[This page](https://mobx.js.org/observable-state.html) says that when using `makeAutoObservable`, "All functions on prototype become `autoAction`." But what does this mean?

If one Googles "MobX autoAction" the top result is [this page](https://mobx.js.org/actions.html) which does not **appear** to mention `autoAction` — well, actually it does, but a normal Ctrl+F won't find it! This is because the only mention of `autoAction` is hidden inside a tiny "note" which must be clicked to expand.

There should be a page specifically to document the modes available to `makeAutoObservable` (and to other functions that support `autoAction`), and the documentation of `makeAutoObservable` (et al) should link to it.

**Comment by qwertie** [comment]
It turns out that the documentation I was looking for already exists [elsewhere](https://mobx.js.org/observable-state.html#available-annotations) on the page describing `makeAutoObservable`.

I would still suggest using links, and changing the description of autoAction from "methods that can act as action or derivation, based on their calling context" to "methods that can act as `action` or `computed`, based on their calling context" and then explain how MobX detects the "calling context".

**Comment by urugator** [comment]
`action` wraps the function with `untracked` to prevent accidental subscriptions. However in case of *auto*  behavior, we don't know whether a function is intended for mutating state or as a helper method for reading state ("derivation") inside the tracking context (`observer`, `autorun`, `reaction`). So we introduced `autoAction` that allows both: it applies *batching* like `action`, it works with `enforceAction` etc, but it doesn't apply `untracked`, which allows for subscriptions. It never acts as `computed`.

Ragarding doc improvements, PR welcome :)
